### ISSUE TITLE
Add a tokenizer/lexer

### ISSUE THREAD

**Issue by lazarljubenovic** [dsherret/ts-simple-ast#502]
**Is your feature request related to a problem? Please describe.**

I'm trying to build a syntax highlighter.

**Describe the solution you'd like**

A tokenize method which would return an array of tokens with LOC information, so that the original code can be reconstructed as-is, with correct whitespace and comments.

**Describe alternatives you've considered**

- [Esprima](http://esprima.org/demo/parse.html#) (see the last tab on the right) doesn't work with TypeScript as it's a JavaScript syntax error.
- [TypeScript itself](https://github.com/Microsoft/TypeScript/wiki/Using-the-Compiler-API) doesn't even mention lexems or tokens in this context.

**Comment by dsherret** [comment]
Yeah, as you mentioned it would probably be best to use the scanner. See here for another example:

https://github.com/dsherret/ts-simple-ast/issues/499#issuecomment-441828794
